qiskit/aqua/algorithms/minimum_eigen_solvers/qaoa/var_form.py

```diff
@@ -90,7 +96,18 @@ def construct_circuit(self, parameters, q=None):
 
         for idx in range(self._p):
             circuit = (self._cost_operator * parameters[idx]).exp_i().compose(circuit)
-            circuit = (self._mixer_operator * parameters[idx + self._p]).exp_i().compose(circuit)
+            if isinstance(self._mixer, OperatorBase):
+                mixer = cast(OperatorBase, self._mixer)
+                circuit = (mixer * parameters[idx + self._p]).exp_i().compose(circuit)
+            else:
+                # mixer as a quantum circuit that can be parameterized
+                mixer = cast(QuantumCircuit, self._mixer)
+                num_params = mixer.num_parameters
+                param_values = parameters[self._p + num_params * idx:
+                                          self._p + num_params * (idx + 1)]
```

Comment:
maybe add a comment that the first `[:self._p]` parameters are used for the cost operator, while the remaining `[self._p:]` parameters are used for the mixer (and that there might be multiples per layer, depending on the circuit).

Reply:
You right, added a few comments.